docs/concepts/endpoint.md

```diff
@@ -0,0 +1,38 @@
+---
+id: endpoint
+title: Endpoint
+---
+
+Endpoints in ZIO HTTP represent individual API operations or routes that the server can handle. They encapsulate the structure and behavior of the API endpoints in a type-safe manner, making them fundamental building blocks for defining HTTP APIs.
+
+## Endpoint Definition
+
+Endpoints are defined using the `Endpoint` object's combinators, such as `get`, `post`, `path`, `query`, and more. These combinators allow us to specify various aspects of the endpoint, such as the HTTP method, URL path, query parameters, request/response bodies, and more.
+
+## Middleware
+
+Middleware can be applied to endpoints using the `@@` operator. Middleware allows us to add additional behavior or processing to the endpoint. For instance, we can handle authentication, validation, error handling, logging, or implement any custom logic needed for the endpoint using middleware.
```

Comment:
It is a good practice to enclose the data types inside backticks such as `Middleware`, `Response`, `Request`, `Route`, etc., wherever required throughout the entire documentation.